test/src/main/java/org/apache/accumulo/test/ample/usage/TabletFileUpdateIT.java

```diff
@@ -0,0 +1,220 @@
+import org.apache.accumulo.harness.SharedMiniClusterBase;
+import org.apache.accumulo.server.tablets.TabletTime;
+import org.apache.accumulo.test.ample.metadata.TestAmple;
+import org.apache.accumulo.tserver.MinorCompactionReason;
+import org.apache.hadoop.fs.Path;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests tablet usage of ample to add a new minor compacted file to tablet. This tests edge cases,
+ * the normal cases are well tested by many other ITs from simply running Accumulo.
+ */
+public class TabletFileUpdateIT extends SharedMiniClusterBase {
+
+  @BeforeAll
+  public static void setup() throws Exception {
+    SharedMiniClusterBase.startMiniCluster();
+  }
+
+  @AfterAll
+  public static void teardown() {
+    SharedMiniClusterBase.stopMiniCluster();
+  }
+
+  private static final ReferencedTabletFile newFile =
+      ReferencedTabletFile.of(new Path("file:///accumulo/tables/t-0/b-0/F1.rf"));
+  private static final DataFileValue dfv1 = new DataFileValue(1000, 100);
+  private static final TServerInstance tserverInstance =
+      new TServerInstance("localhost:9997", 0xabcdef123L);
+  private static final TableId tableId = TableId.of("99");
+  private static final KeyExtent extent = new KeyExtent(tableId, null, null);
+
+  /**
+   * This test ensures that a tablet handles tablet time changing externally (like bulk import could
+   * change tablet time).
+   */
+  @Test
+  public void testTimeChanged() throws Exception {
+    String[] tableNames = getUniqueNames(1);
+    String metadataTable = tableNames[0];
+
+    var tabletTime = TabletTime.getInstance(new MetadataTime(1, TimeType.LOGICAL));
+    long flushNonce = 42L;
+
+    try (var client = Accumulo.newClient().from(getClientProps()).build()) {
+      client.tableOperations().create(metadataTable);
+      TestAmple.create(getCluster().getServerContext(),
+          Map.of(Ample.DataLevel.USER, metadataTable));
+
+      Ample testAmple = TestAmple.create(getCluster().getServerContext(),
+          Map.of(Ample.DataLevel.USER, metadataTable));
+
+      assertNull(testAmple.readTablet(extent));
+
+      // create tablet in the fake metadata table
+      testAmple.mutateTablet(extent).putDirName("dir1").putTime(tabletTime.getMetadataTime())
+          .putLocation(Location.current(tserverInstance)).putPrevEndRow(null).mutate();
+
+      // get the tablets metadata
+      var lastMetadata = testAmple.readTablet(extent);
+
+      // update the tablets time, so it will differ
+      testAmple.mutateTablet(extent).putTime(tabletTime.getMetadataTime(tabletTime.getTime() + 2))
+          .mutate();
+
+      tabletTime.updateTimeIfGreater(tabletTime.getTime() + 6);
+
+      updateTabletDataFile(testAmple, tabletTime.getTime(), newFile, dfv1, Set.of(), 7L,
+          MinorCompactionReason.SYSTEM, tserverInstance, extent, lastMetadata, tabletTime,
+          flushNonce);
+
```

Comment:
Is there a case where the tablet's time has changed in the metadata table, maybe from some other operation performed in the Manager? This looks like it tests the positive case, is there a negative case?

Reply:
Yeah there is a good negative case to test.  Added that in 74b8f14